File: src/#21/solution.go
package solution

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
	if l1 == nil {
		return l2
	}
	if l2 == nil {
		return l1
	}
	node := &ListNode{}
	if l1.Val > l2.Val {
		node.Val = l2.Val
		node.Next = mergeTwoLists(l1, l2.Next)
	} else {
		node.Val = l1.Val
		node.Next = mergeTwoLists(l1.Next, l2)
	}
	return node
}

File: src/#21/solution.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val
  this.next = next === undefined ? null : next
}

var mergeTwoLists = function (l1, l2) {
  if (l1 === null) {
    return l2
  }
  if (l2 === null) {
    return l1
  }
  let node = new ListNode()
  if (l1.val > l2.val) {
    node.val = l2.val
    node.next = mergeTwoLists(l1, l2.next)
  } else {
    node.val = l1.val
    node.next = mergeTwoLists(l1.next, l2)
  }
  return node
}
